analysis.go

```diff
@@ -112,13 +112,15 @@ func ListPackages(fileRoot, importRoot string) (PackageTree, error) {
 		// would have an err with the same path as is called this time, as only
 		// then will filepath.Walk have attempted to descend into the directory
 		// and encountered an error.
-		_, err = os.Open(wp)
+		var f *os.File
+		f, err = os.Open(wp)
 		if err != nil {
 			if os.IsPermission(err) {
 				return filepath.SkipDir
 			}
 			return err
 		}
+		defer f.Close()
```

Comment:
Better to just close it straightaway, no `defer`. The only purpose here is to check readability of the dir; we don't actually _do_ anything with it. (Which is why I forgot to close the file in the first place 😞 )